util/dataset.go

```diff
@@ -191,6 +202,11 @@ func (d *DataSet) Validate() error {
 	return nil
 }
 
+func (d *DataSet) ValidType() bool {
```

Comment:
Same question here: can this method be package visible?

Reply:
yes, changed